Python/Product/PythonTools/PythonTools/Editor/AutoIndent.cs

```diff
@@ -84,7 +84,10 @@ private struct LineInfo {
                 }
 
                 string sline = tokens[tokenIndex].Span.GetText();
-                var lastChar = sline.Length == 0 ? '\0' : sline[sline.Length - 1];
+                char lastChar = '\0';

```

Comment:
It avoids a Roslyn bug that generates invalid code and causes InvalidProgramExceptions.